### PR TITLE
fix: fail when cannot get token, add Akash to generateText switch

### PR DESCRIPTION
**Pull request by vpavlin**
# Relates to:

# Risks

* this throws when we cannot get a token based on provider (otherwise it silently runs and prints auth errors) - it is a change in behaviour

# Background

## What does this PR do?

This fixes 2 issues I've hit:
* When I setup Akash provider I got `Unsupported provider error` because it was missing in `generateText` switch
* When I was trying to update eliza-starter and use it with Akash or Venice, I'd get auth errors because it could not get the provider token - since the `getTokenForProvider` did not account for the new providers, but it would not fail there, hence adding a `default` case which throwa

cc @MbBrainz

## What kind of change is this?

Bug fixes

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?

* You can try to use Akash provider and chat with the agent - it should result in `Unsupported provider error` without this change
* You can also try to configure some unknown provider and gcall `getTokenForProvider` andthe function will not report any errors without this change 

## Discord username
@vpavlin

### CODE REVIEW

**Comment by vpavlin** [comment]
Wait, should I use `develop` or `main` as a base?

**Comment by monilpat** [comment]
> Wait, should I use `develop` or `main` as a base?

Develop :)

File: packages/core/src/generation.ts [hunk 1]
```diff
@@ -176,7 +176,8 @@ export async function generateText({
             case ModelProviderName.LLAMACLOUD:
             case ModelProviderName.NANOGPT:
             case ModelProviderName.HYPERBOLIC:
-            case ModelProviderName.TOGETHER: {
+            case ModelProviderName.TOGETHER:
```
Comment (monilpat) [inline packages/core/src/generation.ts:179]: Please add akash to the other generateX methods as relevant

**Review by monilpat — CHANGES_REQUESTED** [review]
LGTM outside of comment

File: packages/core/src/generation.ts (hunk 1 above)
Comment (shakkernerd) [reply packages/core/src/generation.ts:179]: Added to handleProvider function.

**Review by shakkernerd — APPROVED** [review]

**Comment by shakkernerd** [comment]
> LGTM outside of comment

This is done

**Comment by vpavlin** [comment]
:pray:

**Pull request merged**